### ISSUE TITLE
Unable to authenticate to local BIMServer from BIMSurfer, Windows 7

### ISSUE THREAD

**Issue by powerslave** [opensourceBIM/BIMsurfer-before2019#68]
SImilar to Issue #66, but on Windows 7

BIMServer Version: 1.0 (03-02-2011 21:39) runs on localhost:8082. I can login as admin, or with my created user with no problems.

When I try to connect to BIMServer from BIMSurfer (v 46a70b9) with settings
http://localhost:8082, my username (email) and password, I get **Login request failed**

I tried this with same results in the following configurations:

BIMSurfer on Chrome 16.0.912.63 m, with --allow-file-access-from-files --disable-web-security
- as local file (file:///D:/www/bimserver-BIMsurfer-46a70b9/index.html#)
- on local Apache http://localhost:8080/bimserver-BIMsurfer-46a70b9/#

Firefox 8 with security.fileuri.strict_origin_policy set to false
- as local file
- on local apache

I tried all combinations with both my created user, as well as with the admin (admin@bimserver.org, pw: admin)

Is there anything obvious I am doing wrong?

**Comment by powerslave** [comment]
(Don't know if that's helpful)

Chrome Firebug reports the following statement (anonymified)
GET http://localhost:8082/login.jsp?username=USER%40MYSERVER.COM&password=MYPASSWORD

Headers
ResponseHeaders: EMTPY
RequestHeaders: Accept  _/_

Response
EMPTY

Any other logs or error messages that are helpful?

**Comment by powerslave** [comment]
Same tests now with BIMServer 1.1 (bimserver-combined-1.1.0-2012-01-05.jar)

Same results: **Login request failed** when connecting from BIMSurfer

**Comment by timocontecht** [comment]
Login failed also using Chrome on a Ubuntu Oneiric. I can log-in but it does not show the projects on my server. The BIM server runs on an external server machine.

**Comment by cakewalker** [comment]
I'm having the same issue - 'Login Request Failed'. I've tried it on a number of computers with different setups:
- Windows 7 with BIM Server running within Tomcat
- Windows XP with BIM Server running within Tomcat (tried on 2 different PCs)
- Windows XP with BIM Server run from the JAR file (tried on the same 2 PCs, did try disabling Tomcat and also using a different port to Tomcat)
- Ubuntu with BIM Server run from the JAR file

All are running BIM Server 1.1, although I did have 1.0 running (then undeployed) within Tomcat originally. I've tried a variety of builds of 1.1, sorry but I can't say which one in which environment. I've checked firewall settings, which appear to be letting things through. On the XP machines when running the JAR file I couldn't get into the BIM Server interface. I can access the BIM Server interface on the one running on Ubuntu (and can access it from other PCs on the network - still get the Login Request error when accessing from BIM Surfer though). The version running on Ubuntu is bimserver-combined-1.1beta-2011-11-18 - I've not tried any other versions on this computer. In all instances, BIM Surfer is running in an Apache web server. I've tried accessing it from up-to-date versions of both Firefox and Chrome.

I'm running out of ideas for permutations to try - any suggestions would be very welcome.

**Comment by cakewalker** [comment]
After a bit more trying I think I've found a workaround. Sorry if the terminology I use isn't quite correct - I'm not that familiar with the workings of AJAX, SOAP etc.

I think the issue is to do with BIM Surfer needing to authenticate to BIM Server across addresses, something which I understand AJAX doesn't allow (by default). On all my setups - and I think on those above and on Issue #66 - BIM Surfer is on a webserver running on a different port to BIM Server (eg., BIM Surfer on Apache on port 80, BIM Server on Tomcat on port 8080). Putting BIM Surfer in the same place as BIM Server, so that it is served from the same IP address and port, worked.

I am now able to authenticate to the BIM Server and get a list of projects. Unfortunately, trying to 'open' a project (pull the SceneJS data from BIM Server) gives a 500 error (can be seen in Chrome's log).

**Comment by smeza** [comment]
How did you manage to get both BIM server and surfer to run at same port.

**Comment by cakewalker** [comment]
I've been using Tomcat - all I did was copy the BIMsurfer folder into the BIMserver webapp folder. You will have to specifically type index.html in the browser as Tomcat won't know to open index.html by default.

**Comment by smeza** [comment]
I have simply launched bimserver-combined-1.1.0-RC3.jar, set it to run at port 8082 and it works great on a remote web server. As you all I also used Apache set it to run at port 80, uploaded BIMsurfer and I have no problem to access it from any browser.

As you said I should copy BIMsurfer folder into the BIMserver webapp folder. 
My question here is where can I find this folder.

Sebastjan

**Comment by cakewalker** [comment]
Hi Sebastjan,

The JAR version of BIMserver should I think have decompressed into a folder of the same name. Inside that there is a www folder - try copying the BIMsurfer folder into that. I'm sure this isn't the 'correct' way to go about doing this but it should work! BIMsurfer should therefore be at http://localhost:8082/bimsurfer/index.html

Please note that I've managed to get a list of BIMserver projects in BIMsurfer by doing this, but am still unable to open any of those projects. Trying to open any BIMserver project results in a 500 error (in the web browser's console - the BIMsurfer page itself doesn't display any error). I think this is a BIMserver fault, you also get it when typing the REST address and query into the browser's address bar and you also get it regardless of whether it is SceneJS data requested (which is what BIMsurfer requests) or IFC data - in other words the problem affects more than just the SceneJS BIMserver plugin. I've mentioned it to BIMserver's developers.

I understand that, if you were using Tomcat and Apache, it's possible to get the two working together to some extent (perhaps leading to Tomcat apps being served through Apache's port?). I've not delved into how to do that yet but I imagine this would be preferable to the arrangement described above.

Regards,
Chris

**Comment by yllchan** [comment]
Is there any workaround for the localhost?

I use locahost:8082 for BIMServer and localhost:8000 for BIMSurfer

**Comment by cakewalker** [comment]
Hi yllchan,

You mean is there any workaround to them being on different ports/domains? I think the issue of having to have BIMserver and BIMsurfer at the same domain and port is due to browser security - it's my understanding that browsers by default won't let Javascript on a webpage at one domain/port retrieve data from a different domain/port. Then again, there must be a way around this that allows Javascript libraries to pull data from APIs such as Twitter's - sorry but I don't know how they achieve that, though.

Chris

yllchan wrote:

**Comment by hoogia** [comment]
to cakewalker:

did you solved this problem? I did the same as you

**Comment by cakewalker** [comment]
Hi hoogia,

Are you aware of the latest versions of BIMserver having BIMsurfer bundled?

**Comment by hoogia** [comment]
To cakewalker:
Thanks for replying, do you mean by the version of BIMserver1.1.0-2012-06-24 on http://code.google.com/p/bimserver/downloads/list?
I saw it added a "Browser" function, but I can't find the showing photoes like surfer does.

**Comment by cakewalker** [comment]
I think whichever is the latest - the browser function has been part of BIMserver for a month or so now.

To get to it you have to browse to your project (in BIMserver) and type 'surf' - there's no box to type in, just type it. Unfortunately, you'll still run into problems as BIMserver no longer seems to support the method BIMsurfer uses to load project data. Someone has provided a fix for this (scroll to the bottom of the discussion):

http://support.bimserver.org/bimserver/topics/bim_server_surfer_troubleshooting_500_error_when_trying_to_download_rest_data

Basically, you need to download https://dl.dropbox.com/u/580643/BIMsurfer/app.js and use it to replace app.js in the BIMsurfer that's in BIMserver (if you are using the BIMsurfer you've downloaded separately from github you'll need to replace app.min.js with a renamed copy of the new app.js)

nb., the 'surf' command only seems to work when you're using Chrome - I've also tried in Firefox and nothing seems to happen.

**Comment by hoogia** [comment]
@cakewalker 
It's my mistake, indeed the browser function has been exist before.

According to your description and my understanding, here is my steps:
1.Download the new app.js to replace the old app.js, and rename the copy of new app.js into app.min.js, then replace the old app.min.js in BIMSurfer
2. Put Surfer folder in Server's server/www/ folder 
3. After running Server, and Surfer can be opened on 8080/surfer/index.html, by the way, I used Chrome as you said.
4. But still the same situation, it can list projects' name and cannot open or show graph

Is there anything wrong with my operation? And so much grateful to you.

**Comment by hoogia** [comment]
@cakewalker 
After I used Win7, and with .jar, it's solved! But tomcat in Win7 can just log in and no project list showed, it's too wired..
Anyway, it's done, thank you very much for your help:)

**Comment by Siddle111** [comment]
Hi Everybody,

I am new user. I have install tomcat on ubuntu and i plugin BIM Surfer on tomcat but i can't reset or login it has default email and password. i can't reset it. 

I wish anybody will help me to reset log and password. as i can log perfectly.
Thanks,

**Comment by rubendel** [comment]
@Siddle111 This issue is 4 years old, please create a new issue and clearly state which version you are using.

**Comment by Siddle111** [comment]
@rubendel 
I am using BIM surfer 3.0. it shows loading BIM views....... 
How can i access BIM surfer Interface.
thanks
